### PR TITLE
Add GPS supports

### PR DESCRIPTION
**Pull request by viethung7899**
### GPS supports

- Allow technician to submit GPS data
- Show maps and locations with Mapbox API (access token required)

### Other

- Simplify the form wizard

### CODE REVIEW

**Comment by chrishappy** [comment]
Question: to clarify, can we track the GPS coordinates automatically?

We would need the technicians to grant us permission to access their location initially, but afterwards, we would like to require a GPS coordinate without needing any action from the technician.

File: components/forms/farmer-journal/Confirmation.tsx
Comment (chrishappy) [inline components/forms/farmer-journal/Confirmation.tsx:1]: I like the formatting!

File: components/forms/preview/LocationPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { useTranslations } from 'next-intl';
+import { Map, Marker } from 'react-map-gl';
+
+const ACCESS_TOKEN = process.env.NEXT_PUBLIC_MAPBOX_ACCESS_TOKEN
+
+export const LocationPreview = ({ location }: { location: Location }) => {
+  return <>
+    <Map
+      mapLib={import('mapbox-gl')}
+      mapStyle="mapbox://styles/mapbox/streets-v9"
```
Comment (chrishappy) [inline components/forms/preview/LocationPreview.tsx:13]: Could we make the style a constant?

File: components/forms/preview/LocationPreview.tsx
Comment (chrishappy) [inline components/forms/preview/LocationPreview.tsx:1]: I'll need to check with Mr. Jamane whether he wants the technicians to preview the location or not.

File: components/forms/technician-visit/Note.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { useTranslations } from "next-intl"
 export default function FarmerNote() {
   const { register } = useFormContext<TechnicianVisitFormSchema>()
   const t = useTranslations("FarmChecklist")
-  return <div>
+  return <div className="p-4">
```
Comment (chrishappy) [inline components/forms/technician-visit/Note.tsx:8]: Love the attention to detail :)

File: lib/offline/db.ts [hunk 3]
```diff
@@ -21,8 +23,8 @@ export class OvoflowDexie extends Dexie {
 
   constructor() {
     super(DATABASE_NAME);
-    this.version(1).stores({
-      technicianVisit: '++id, value, farmerId',
+    this.version(2).stores({
```
Comment (chrishappy) [inline lib/offline/db.ts]: Could we make the version number a constant?

File: messages/pt.json
Comment (chrishappy) [inline messages/pt.json:1]: @JoelJuniorManuelFombe Could you review the translation?

**Review by chrishappy — APPROVED** [review]

File: components/forms/preview/LocationPreview.tsx (hunk 1 above)
Comment (viethung7899) [reply components/forms/preview/LocationPreview.tsx:13]: I don't think it's necessary. The `mapstyle` is a part of Mapbox documentation.

File: lib/offline/db.ts (hunk 3 above)
Comment (viethung7899) [reply lib/offline/db.ts]: Yeah, I can do that

**Comment by viethung7899** [comment]
Technicians only need to grant permission once to get the GPS.

However, GPS is only called once when the technician form renders, and can be called again in case of error.

Alternatively, I can make the GPS automatically updates as the users move.

What do you think? @chrishappy

**Comment by chrishappy** [comment]
> However, GPS is only called once when the technician form renders, and can be called again in case of error.
> 
> Alternatively, I can make the GPS automatically updates as the users move.

I think we can just store the location upon rendering for now.

We can talk to Mr. Jamene a bit more about this feature.

**Comment by viethung7899** [comment]
Sure, I gonna merge this PR now

**Pull request merged**